packages/modular-scripts/src/build.ts

```diff
@@ -0,0 +1,619 @@
+      `package.json at ${packagesRoot}/${directoryName} does not have a valid "version", bailing...`,
+    );
+    return false;
+  }
+
+  if (packageJson.module) {
+    console.error(
+      `package.json at ${packagesRoot}/${directoryName} shouldn't have a "module" field, bailing...`,
+    );
+    return false;
+  }
+
+  if (packageJson.typings) {
+    console.error(
+      `package.json at ${packagesRoot}/${directoryName} shouldn't have a "typings" field, bailing...`,
+    );
+    return false;
+  }
+
+  console.log(`building ${packageJson.name} at packages/${directoryName}...`);
+
+  const bundle = await rollup.rollup({
+    input: path.join(packagesRoot, directoryName, packageJson.main),
+    external: (id) => {
+      // via tsdx
+      if (id === 'babel-plugin-transform-async-to-promises/helpers') {
+        // we want to inline these helpers
+        return false;
+      }
+      // exclude any dependency that's not a realtive import
+      return !id.startsWith('.') && !path.isAbsolute(id);
+    },
+    treeshake: {
+      // via tsdx: Don't use getters and setters on plain objects.
+      propertyReadSideEffects: false,
+    },
+    plugins: [
+      resolve({
+        extensions,
+        browser: true,
+        mainFields: ['module', 'main', 'browser'],
+      }),
+      commonjs({ include: /\/node_modules\// }),
+      babel({
+        babelHelpers: 'bundled',
+        presets: [
+          ['@babel/preset-typescript', { isTSX: true, allExtensions: true }],
+          '@babel/preset-react',
+          [
+            '@babel/preset-env',
+            {
+              // targets: { esmodules: true }, // Use the targets that you was already using
+              // bugfixes: true, // will be default in Babel 8
+            },
+          ],
+        ],
+        plugins: ['@babel/plugin-proposal-class-properties'],
+        extensions,
+        include: [`${packagesRoot}/**/*`],
+        exclude: 'node_modules/**',
+      }),
+      postcss({ extract: false, modules: true }),
+      // TODO: add sass, css modules, dotenv
+      json(),
+      {
+        // via tsdx
+        // Custom plugin that removes shebang from code because newer
+        // versions of bublé bundle their own private version of `acorn`
+        // and I don't know a way to patch in the option `allowHashBang`
+        // to acorn. Taken from microbundle.
+        // See: https://github.com/Rich-Harris/buble/pull/165
+        name: 'strip-shebang',
+        transform(code) {
+          code = code.replace(/^#!(.*)/, '');
+
+          return {
+            code,
+            map: null,
+          };
+        },
+      },
+    ],
+    // TODO: support for css modules, sass, dotenv,
+    // and anything else create-react-app supports
+    // (alternatively, disable support for those in apps)
+  });
+
+  const outputOptions = {
+    freeze: false,
+    sourcemap: true, // TODO: read this off env
+    globals: { react: 'React', 'react-native': 'ReactNative' }, // why?
```

Comment:
Agreed - why?

Reply:
will just remove this, doesn't make sense for us.